### DISCUSSION TITLE
[Solved] Send later failed-removal of unsent email

### DISCUSSION THREAD

**Discussion by OrangeHedgehog** [Extended-Thunder/send-later#569] [Q&A] [answered]
I created an email to send later, the addressees were all BCC.  Now the email appears to be in a queue to send, but won't.  Each time I start Thunderbird I'm prompted to send the email, and each time it fails.

![image](https://github.com/Extended-Thunder/send-later/assets/73525855/50d92fb3-c582-4a6e-a974-3d273e113878)

It doesn't help that, for whatever reason, Thunderbird isn't showing the email to be sent in the Drafts folder, and Thunderbird also is not showing an Outbox (which _is_ visible in Outlook).

I created a new delayed email (using only a To: addressee), and that sent okay.  I then created another delayed email, using a BCC: addressee.  This failed the same as for the previous bulk BCC email.  However, whilst the cause may have been identified (Send Later cannot cope with BCC addresses, apparently (until such time as Send Later can cope with BCC addressees it would be useful if Send Later checked for the existence of any BCC addressees and stopped\prevented the email from entering the queue).  

The core question is: **how do I fix Send Later to stop it trying to send these non-existent messages?**  (Neither of them are listed in the Drafts folder.)

Disabling Send Later did not prevent the Unsent Messages dialogue from appearing.
Worse: having removed Send Later, the Unsent Messages dialogue still appeared.
(In both cases I shut-down and restarted Thunderbird.)

Send later version: 9.2.9, 28 Jul 2023.
Thunderbird version: 102.13.1 (32-bit)
Full disclosure: when checking the version of Thunderbird it offered an update, so I did.  Thunderbird is still trying (and failing) to send the non-existent messages.

**Comment by jikamens** [comment, marked as answer]
This has nothing to do with the fact that you used BCC. The problem is that one of the recipient email addresses is invalid; this would have happened if you had put it in To or CC as well.

The bad message is in your Outbox folder under Local Folders. If you didn't find the Outbox folder you looked in the wrong place.

Delete the message from the Outbox folder, or move it from Outbox into Drafts and edit it to fix the problem and then resend it, and the problem will go away.

**Reply by OrangeHedgehog** [reply]
Hi, thank you for the quick reply, it's appreciated.

I have now found the Local\Outbox folder, something I clearly wasn't looking for successfully (with eight email accounts it wasn't easy to spot right at the bottom, completely away from the email account, but hey, I guess Thunderbird does things differently from Outlook and that's just something folks have to get used to - including apparently preventing an email being sent, because under Outlook the email would have sent and then I would have received an error message from the mail server).

Pesky email deleted now, thanks. 👍👍

I don't know if you're able to answer this, but given the presence of even just one invalid email address, would that have prevented the email from being sent to the other recipients, or would those whose email addresses were correct have received the email?  (Because I need to know if I have to resend the email to everyone, also, if the message was sucessfully sent to all the valid email addresses each time, I've now sent it to them multiple times and would need to apologise!)

**Comment by jikamens** [comment]
>under Outlook the email would have sent and then I would have received an error message from the mail server).

I don't know about that. Email delivery can fail to deliver in two different ways: (1) it can fail when your mail client is talking to your mail server (that's what's happening in the case we're discussing here); (2) it can fail when our mail server is attempting to deliver the message to the recipient's mail server. The latter will definitely cause a separate bounce message in either Thunderbird or Outlook. It's _possible_ that Outlook would also generate a bounce message in the former case; but the more correct behavior would be what you're seeing in Thunderbird.

>would that have prevented the email from being sent to the other recipients, or would those whose email addresses were correct have received the email?

This is... complicated.

In failure mode type (1), the message almost certainly was not delivered to any of its recipients.

In failure mode type (2), the message is usually delivered to all recipients who are on mail servers other than the mail server for the bad recipient, and may or may not be delivered to the good recipients on the bad recipient's mail server, depending on how _your_ mail server behaves. If it attempts to group multiple recipients on the same destination server into a single delivery (the more correct behavior), probably none of them got it. If it delivers to each recipient in a separate transaction even for recipients on the same destination server, then all of them would have received it except for the bad recipient. Sometimes you can tell which of these is the case depending the exact content of the bounce message you get back.

**Reply by OrangeHedgehog** [reply]
As I haven't received any emails from any of the intended recipients (either acknowledging what was sent or complaining about having received it more than once), and nor have I received bounce messages from the mail servers, I think, and hope, perhaps "type (1)" happened.  Thankfully! 

I had also sent an immediate send email (using Outlook, as a reply to another email with the same recipients), which apparently included the failed addressees.  Sending that resulted in a _System Administrator_ email being received with the invalid email addresses being specifically identified.

Having tried to manually send the BCC email (using Outlook), I received a separate _System Administrator_ email.  This time listing _all_ of the email addresses, and citing an Error 554, so hopefully this is more evidence to suggest the email didn't go out at all.  Well, not that one anyway, as that was Outlook and not Thunderbird and Send Later.  I checked the link provided in the email and the Website listed these five scenarios:

1. 554 Invalid DNS PTR resource record
2. 554 Bad DNS PTR resource record
3. 554 IP address is blacklisted
4. 554 Reject due to policy restrictions
5. 554 Transaction failed, reject due to policy restrictions

Disappointingly within the _System Administrator_ email all it said was that each of the recipent email addresses failed, without clarification as to which specific scenario.

An extract of the text is, "Your message did not reach some or all of the intended recipients.

      Subject:	« redacted»
      Sent:	27/07/2023 13:18

The following recipient(s) cannot be reached:

      '«email address or Undisclosed Recipients»' on 27/07/2023 13:19
            554 For explanation visit «URL»
"
In each case the URL was identical.

I didn't think this was related to the Send Later issue though, hence not mentioned it previously.

Thank you again for your quick reply.
